src/DataValues/DecimalValue.php

```diff
@@ -58,7 +58,9 @@ public function __construct( $value ) {
 		}
 
 		$value = trim( $value );
-
+		if ( $value !== '' && $value[0] !== '-' && $value[0] !== '+' ) {
```

Comment:
I don't think this is the best implementation. It will add plus signs in front of strings that don't start with a number. It will, for example, produce `"+−1"` when the string starts with an actual minus sign. What was wrong with my suggestion?